plugins/database/postgresql/postgresql.go

```diff
@@ -501,3 +516,48 @@ func (p *PostgreSQL) RotateRootCredentials(ctx context.Context, statements []str
 	p.RawConfig["password"] = password
 	return p.RawConfig, nil
 }
+
+// containsMultilineStatement is a best effort to determine whether
+// a particular statement is multiline, and therefore should not be
+// split upon semicolons. If it's unsure, it defaults to false.
+func containsMultilineStatement(stmt string) bool {
+	// We're going to look for the word "END", but first let's ignore
+	// anything the user provided within single or double quotes since
+	// we're looking for an "END" within the Postgres syntax.
+	literals, err := extractQuotedStrings(stmt)
+	if err != nil {
+		return false
+	}
+	stmtWithoutLiterals := stmt
+	for _, literal := range literals {
+		stmtWithoutLiterals = strings.Replace(stmt, literal, "", -1)
+	}
+	// Now look for the word "END" specifically. This will miss any
+	// representations of END that aren't surrounded by spaces, but
+	// it should be easy to deal with. This approach was chosen to
+	// avoid accidently matching with unintended words like "APPEND".
+	return strings.Contains(stmtWithoutLiterals, " END ")
```

Comment:
You could use a regexp to look for only END surrounded by word boundaries, i.e. `\bEND\b`.

Reply:
Ah! I wasn't aware of that, thanks! Will do.

Reply:
Done in https://github.com/hashicorp/vault/pull/8512/commits/645789ba3e570f403a8a33424a70468a34ed4db8